### ISSUE TITLE
Oracle.EntityFrameworkCore update statement and checking rows affected

### ISSUE THREAD

**Issue by SierraNL** [oracle/dotnet-db-samples#180]
When creating an update the following SQL is generated:
```sql
DECLARE
v_RowCount INTEGER;
BEGIN
UPDATE "MyTable" SET "Processed" = :p0, "Updated" = :p1
WHERE "Id" = :p2 AND "Type" = :p3 AND "MessageId" = :p4;
v_RowCount := SQL%ROWCOUNT;
OPEN :cur1 FOR SELECT v_RowCount FROM DUAL;
 
END;
```

This code is not efficient because it requires context switching and setting up a new cursor.

is there no other way within the framework that can process the return value of the UPDATE statement?

**Comment by alexkeh** [comment]
If Microsoft introduces a SaveChanges API in EF Core that doesn't require returning the number of rows to the database, then neither Oracle nor any other provider writer would have to retrieve this info.

**Comment by SierraNL** [comment]
Ok clear, but isn't it possible to return the SQL%ROWCOUNT without the cursor?

**Comment by alexkeh** [comment]
Good idea! We'll review whether this idea works. Thanks for the suggestion.

**Comment by alexkeh** [comment]
@SierraNL Good suggestion! We'll review this possible change to improve performance.

I've created a formal enhancement request (33050516) within Oracle's bug/ER tracking system to have a developer try this out.

**Comment by alexkeh** [comment]
The fix will be for EF Core 10 and higher. EF Core 8 and 9 are nearing EOL.
